runtime/src/bank.rs

```diff
@@ -866,9 +870,15 @@ impl Bank {
         });
 
         // Following code may touch AccountsDB, requiring proper ancestors
-        if parent.epoch() < new.epoch() {
+        let parent_epoch = parent.epoch();
+        if parent_epoch < new.epoch() {
             new.apply_feature_activations(false);
         }
+
+        if new.enable_stake_rewrite(parent_epoch) {
+            // Remove me after a while around v1.6
+            new.rewrite_stakes();
+        }
```

Comment:
It's not immediately obvious to me why this can't be part of `apply_feature_activations`

Reply:
yeah, a bit of understanding of implied subtlety is needed. In short, `apply_feature_activations` is for proper state restoration of _non-AccountsDB `Bank` fields_. Here, we're rewriting AccountsDB state (stake accounts). `apply_feature_activation` is called from both normal child bank creation and both from snapshot restore.

I'll add a comment for this.

Reply:
`reconfigure_token2_native_mint()` is in `apply_feature_activations`... is the difference there the capitalization adjustment?
Anyway, I think a comment would help, thanks!

Reply:
hmm. you're right in this regard as well.. I've fixed https://github.com/solana-labs/solana/pull/13461/commits/a35d8ded5510834464f94ff5759a8e79c1eb1f50.

turned out we only needed special considerations for builtins.

Reply:
Nice! So much simpler :)